internal/devpkg/package.go

```diff
@@ -100,12 +102,14 @@ func PackagesFromStringsWithOptions(rawNames []string, l lock.Locker, opts devop
 	return packages
 }
 
-func PackagesFromConfig(config *devconfig.Config, l lock.Locker) []*Package {
+func PackagesFromConfig(packages []configfile.Package, l lock.Locker) []*Package {
 	result := []*Package{}
-	for _, cfgPkg := range config.Packages() {
-		pkg := newPackage(cfgPkg.VersionedName(), cfgPkg.IsEnabledOnPlatform(), l)
+	for _, cfgPkg := range packages {
+		pkg := newPackage(cfgPkg.VersionedName(), cfgPkg.IsEnabledOnPlatform, l)
 		pkg.DisablePlugin = cfgPkg.DisablePlugin
-		pkg.PatchGlibc = cfgPkg.PatchGlibc && nix.SystemIsLinux()
+		pkg.patchGlibc = sync.OnceValue(func() bool {
+			return cfgPkg.PatchGlibc && nix.SystemIsLinux()
```

Comment:
is changing to a function _really_ needed?

I'm not convinced because:`nix.SystemIsLinux` relies on `nix.System` which is pre-computed once when running the cobra command, so likely won't be an actual system call here

Reply:
This is needed because unfortunately devbox CLI calls `devbox.Open` when populating autocomplete for scripts. (https://github.com/jetpack-io/devbox/blob/main/internal/boxcli/run.go#L54-L54). Note that this code happens before `ensureNixInstalled` is called. Arguably, we don't want to require nix for stuff like listing scripts which doesn't need nix.

So `devbox.Open` cannot call anything requiring nix, otherwise all commends will fail. Without this change, just parsing a devbox.json requires nix. 

We should probably replace `command.ValidArgs` with `command.ValidArgsFunction` but even then, I think it's useful to keep `devbox.Open` working without nix.